analyzers/src/SonarAnalyzer.CSharp/Rules/SqlKeywordsDelimitedBySpace.cs

```diff
@@ -140,25 +142,17 @@ public override void VisitBinaryExpression(BinaryExpressionSyntax node)
                     && TryGetStringWrapper(node.Right, out var rightSide)
                     && StartsWithSqlKeyword(leftSide.Text.Trim()))
                 {
-                    var strings = new List<StringWrapper>();
-                    strings.Add(leftSide);
-                    strings.Add(rightSide);
-                    var onlyStringsInConcatenation = AddStringsToList(node, strings);
-                    if (!onlyStringsInConcatenation)
+                    var strings = new List<StringWrapper> { leftSide, rightSide };
+                    if (TryExtractNestedStrings(node, strings))
                     {
-                        return;
+                        RaiseIssueIfNotDelimited(strings);
                     }
-
-                    RaiseIssueIfSpaceDoesNotExistBetweenStrings(strings);
-                }
-                else
-                {
-                    Visit(node.Left);
-                    Visit(node.Right);
                 }
+                Visit(node.Left);
+                Visit(node.Right);
```

Comment:
I always visit nested nodes, so that we can raise on expressions like `"hey" + $"{there}{dude}"`.

Before, we would miss the `interpolated` part.